StoreKitUnitTests/TrialOrIntroPriceEligibilityCheckerTests.swift

```diff
@@ -0,0 +1,189 @@
+    func testSK1EligibilityIsCalculatedFromReceiptData() throws {
+        setupSK1()
+        let stubbedEligibility = ["product_id": IntroEligibilityStatus.eligible]
+        mockIntroEligibilityCalculator.stubbedCheckTrialOrIntroductoryPriceEligibilityResult = (stubbedEligibility, nil)
+
+        var completionCalled = false
+        var maybeEligibilities: [String: IntroEligibility]?
+        trialOrIntroPriceEligibilityChecker!.sk1CheckEligibility([]) { (eligibilities) in
+            completionCalled = true
+            maybeEligibilities = eligibilities
+        }
+
+        expect(completionCalled).toEventually(beTrue())
+        let receivedEligibilities = try XCTUnwrap(maybeEligibilities)
+        expect(receivedEligibilities.count) == 1
+    }
+
+    func testSK1EligibilityIsFetchedFromBackendIfErrorCalculatingEligibility() throws {
+        setupSK1()
+        let stubbedError = NSError(domain: RCPurchasesErrorCodeDomain,
+                                   code: ErrorCode.invalidAppUserIdError.rawValue,
+                                   userInfo: [:])
+        mockIntroEligibilityCalculator.stubbedCheckTrialOrIntroductoryPriceEligibilityResult = ([:], stubbedError)
+
+        let stubbedEligibility = ["product_id": IntroEligibility(eligibilityStatus: IntroEligibilityStatus.eligible)]
+        mockBackend.stubbedGetIntroEligibilityCompletionResult = (stubbedEligibility, nil)
+        var completionCalled = false
+        var maybeEligibilities: [String: IntroEligibility]?
+        trialOrIntroPriceEligibilityChecker!.sk1CheckEligibility([]) { (eligibilities) in
+            completionCalled = true
+            maybeEligibilities = eligibilities
+        }
+
+        expect(completionCalled).toEventually(beTrue())
+        let receivedEligibilities = try XCTUnwrap(maybeEligibilities)
+        expect(receivedEligibilities.count) == 1
+    }
+
+    func testSK1ErrorFetchingFromBackendAfterErrorCalculatingEligibility() throws {
+        setupSK1()
+        let stubbedError = NSError(domain: RCPurchasesErrorCodeDomain,
+                                   code: ErrorCode.invalidAppUserIdError.rawValue,
+                                   userInfo: [:])
+        mockIntroEligibilityCalculator.stubbedCheckTrialOrIntroductoryPriceEligibilityResult = ([:], stubbedError)
+
+        mockBackend.stubbedGetIntroEligibilityCompletionResult = ([:], stubbedError)
+        var completionCalled = false
+        var maybeEligibilities: [String: IntroEligibility]?
+        trialOrIntroPriceEligibilityChecker!.sk1CheckEligibility([]) { (eligibilities) in
+            completionCalled = true
+            maybeEligibilities = eligibilities
+        }
+
+        expect(completionCalled).toEventually(beTrue())
+        let receivedEligibilities = try XCTUnwrap(maybeEligibilities)
+        expect(receivedEligibilities.count) == 0
+    }
+
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    func testSK2CheckEligibilityAsync() async throws {
```

Comment:
it'd be cool to do a couple of tests in `BackendIntegrationTest` that use SK Config files, that: 
- check eligibility with SK 2 beforehand on a clean user
- make a purchase
- check eligibility afterwards, ensures no longer eligible, both in SK2 and backend (it won't work for the local receipt parser version since that one's not yet compatible with sk config files)